lib/shadow.ml

```diff
@@ -31,9 +31,21 @@ let sp_flag     = field shadow_t "sp_flag" ulong
 
 let () = seal shadow_t
 
+let ptr_char_to_string p =
```

Comment:
Could we avoid the code duplication by moving this code into the Common module?

Reply:
Common is exported and used, I wanted to keep this as an internal detail